--- /home/ilm/Documents/GitHub/Altirra-4.50-test8-src/src/Kasumi/source/resample_stages_x64.cpp	2025-09-14 23:26:06.532817500 +0200
+++ /home/ilm/Documents/GitHub/Altirra-4.50-test9-src/src/Kasumi/source/resample_stages_x64.cpp	2026-04-15 05:41:51.726761900 +0200
@@ -21,6 +21,8 @@
 #include <vd2/Kasumi/resample_kernels.h>
 #include "resample_stages_x64.h"
 
+#ifndef VD_COMPILER_GCC
+
 extern "C" long VDCDECL vdasm_resize_table_col_SSE2(uint32 *out, const uint32 *const*in_table, const int *filter, int filter_width, uint32 w);
 extern "C" long VDCDECL vdasm_resize_table_row_SSE2(uint32 *out, const uint32 *in, const int *filter, int filter_width, uint32 w, long accum, long frac);
 
@@ -45,6 +47,7 @@
 
 	vdasm_resize_table_col_SSE2((uint32*)dst, (const uint32 *const *)src, (const int *)mFilterBank.data() + filtSize*((phase >> 8) & 0xff), filtSize, w);
 }
+#endif
 
 ///////////////////////////////////////////////////////////////////////////
 
@@ -285,7 +288,7 @@
 	uint32 fastGroups = mNumFastGroups;
 	if (fastGroups) {
 		if (mbUseFastLerp) {
-#ifdef VD_COMPILER_CLANG
+#ifdef VD_COMPILER_CLANG_OR_GCC
 			[&] __attribute__((target("ssse3"))) {
 #endif
 			const __m128i round8 = _mm_set1_epi16(0x40);
@@ -302,7 +305,7 @@
 				rowFilter += 16;
 				dst8 += 8;
 			} while(--fastGroups);
-#ifdef VD_COMPILER_CLANG
+#ifdef VD_COMPILER_CLANG_OR_GCC
 			}();
 #endif
 
@@ -576,8 +579,12 @@
 					for(int i=0; i<w4; i += 4) {
 						__m128i accum = round;
 
-						__assume(ksize > 0);
-						for(unsigned j = 0; j < ksize; j += 2) {
+						accum = _mm_add_epi32(accum, _mm_madd_epi16(_mm_unpacklo_epi8(_mm_unpacklo_epi8(_mm_loadu_si32(src[0] + xoffset), _mm_loadu_si32(src[1] + xoffset)), zero), _mm_shuffle_epi32(_mm_loadu_si32(filter +  0), 0)));
+						accum = _mm_add_epi32(accum, _mm_madd_epi16(_mm_unpacklo_epi8(_mm_unpacklo_epi8(_mm_loadu_si32(src[2] + xoffset), _mm_loadu_si32(src[3] + xoffset)), zero), _mm_shuffle_epi32(_mm_loadu_si32(filter +  4), 0)));
+						accum = _mm_add_epi32(accum, _mm_madd_epi16(_mm_unpacklo_epi8(_mm_unpacklo_epi8(_mm_loadu_si32(src[4] + xoffset), _mm_loadu_si32(src[5] + xoffset)), zero), _mm_shuffle_epi32(_mm_loadu_si32(filter +  8), 0)));
+						accum = _mm_add_epi32(accum, _mm_madd_epi16(_mm_unpacklo_epi8(_mm_unpacklo_epi8(_mm_loadu_si32(src[6] + xoffset), _mm_loadu_si32(src[7] + xoffset)), zero), _mm_shuffle_epi32(_mm_loadu_si32(filter + 12), 0)));
+						accum = _mm_add_epi32(accum, _mm_madd_epi16(_mm_unpacklo_epi8(_mm_unpacklo_epi8(_mm_loadu_si32(src[8] + xoffset), _mm_loadu_si32(src[9] + xoffset)), zero), _mm_shuffle_epi32(_mm_loadu_si32(filter + 16), 0)));
+						for(unsigned j = 10; j < ksize; j += 2) {
 							accum = _mm_add_epi32(accum, _mm_madd_epi16(_mm_unpacklo_epi8(_mm_unpacklo_epi8(_mm_loadu_si32(src[j+0] + xoffset), _mm_loadu_si32(src[j+1] + xoffset)), zero), _mm_shuffle_epi32(_mm_loadu_si32(filter + j*2), 0)));
 						}
 
