--- /home/ilm/Documents/GitHub/Altirra-4.50-test8-src/src/h/vd2/system/atomic.h	2023-07-05 04:39:24.584832600 +0200
+++ /home/ilm/Documents/GitHub/Altirra-4.50-test9-src/src/h/vd2/system/atomic.h	2026-04-15 05:41:51.216410500 +0200
@@ -39,7 +39,7 @@
 	#else
 		return (void *)(sintptr)_InterlockedCompareExchange((volatile long *)(volatile sintptr *)pp, (long)(sintptr)p, (long)(sintptr)compare);
 	#endif
-#elif defined(VD_COMPILER_CLANG)
+#elif defined(VD_COMPILER_CLANG_OR_GCC)
 	return __sync_val_compare_and_swap(pp, compare, p);
 #else
 	#error not implemented
@@ -77,7 +77,7 @@
 	static inline int staticExchange(volatile int *dst, int v) {
 		#if defined(VD_COMPILER_MSVC)
 			return (int)_InterlockedExchange((volatile long *)dst, v);
-		#elif defined(VD_COMPILER_CLANG)
+		#elif defined(VD_COMPILER_CLANG_OR_GCC)
 			return __sync_lock_test_and_set((int *)dst, v);
 		#else
 			#error not implemented
@@ -88,7 +88,7 @@
 	static inline void staticIncrement(volatile int *dst) {
 		#if defined(VD_COMPILER_MSVC)
 			_InterlockedExchangeAdd((volatile long *)dst, 1);
-		#elif defined(VD_COMPILER_CLANG)
+		#elif defined(VD_COMPILER_CLANG_OR_GCC)
 			__sync_fetch_and_add(dst, 1);
 		#else
 			#error not implemented
@@ -99,7 +99,7 @@
 	static inline void staticDecrement(volatile int *dst) {
 		#if defined(VD_COMPILER_MSVC)
 			_InterlockedExchangeAdd((volatile long *)dst, -1);
-		#elif defined(VD_COMPILER_CLANG)
+		#elif defined(VD_COMPILER_CLANG_OR_GCC)
 			__sync_fetch_and_sub(dst, 1);
 		#else
 			#error not implemented
@@ -111,7 +111,7 @@
 	static inline bool staticDecrementTestZero(volatile int *dst) {
 		#if defined(VD_COMPILER_MSVC)
 			return 1 == _InterlockedExchangeAdd((volatile long *)dst, -1);
-		#elif defined(VD_COMPILER_CLANG)
+		#elif defined(VD_COMPILER_CLANG_OR_GCC)
 			return 1 == __sync_fetch_and_sub((int *)dst, 1);
 		#else
 			#error not implemented
@@ -123,7 +123,7 @@
 	static inline int staticAdd(volatile int *dst, int v) {
 		#if defined(VD_COMPILER_MSVC)
 			return (int)_InterlockedExchangeAdd((volatile long *)dst, v) + v;
-		#elif defined(VD_COMPILER_CLANG)
+		#elif defined(VD_COMPILER_CLANG_OR_GCC)
 			return __sync_fetch_and_add((int *)dst, v) + v;
 		#else
 			#error not implemented
@@ -135,7 +135,7 @@
 	static inline int staticExchangeAdd(volatile int *dst, int v) {
 		#if defined(VD_COMPILER_MSVC)
 			return _InterlockedExchangeAdd((volatile long *)dst, v);
-		#elif defined(VD_COMPILER_CLANG)
+		#elif defined(VD_COMPILER_CLANG_OR_GCC)
 			return __sync_fetch_and_add((int *)dst, v);
 		#else
 			#error not implemented
@@ -148,7 +148,7 @@
 	static inline int staticCompareExchange(volatile int *dst, int v, int compare) {
 		#if defined(VD_COMPILER_MSVC)
 			return _InterlockedCompareExchange((volatile long *)dst, v, compare);
-		#elif defined(VD_COMPILER_CLANG)
+		#elif defined(VD_COMPILER_CLANG_OR_GCC)
 			return __sync_val_compare_and_swap((int *)dst, compare, v);
 		#else
 			#error not implemented
@@ -172,7 +172,7 @@
 	void operator|=(int v)	{ _InterlockedOr((volatile long *)&n, v); }		///< Atomic bitwise OR.
 	void operator^=(int v)	{ _InterlockedXor((volatile long *)&n, v); }	///< Atomic bitwise XOR.
 
-#elif defined(VD_COMPILER_CLANG)
+#elif defined(VD_COMPILER_CLANG_OR_GCC)
 
 	void operator&=(int v) {
 		__sync_fetch_and_and(&n, v);
@@ -340,7 +340,7 @@
 			#else
 				return ptr == p ? p : (T *)_InterlockedExchange((volatile long *)&ptr, (long)p);
 			#endif
-		#elif defined(VD_COMPILER_CLANG)
+		#elif defined(VD_COMPILER_CLANG_OR_GCC)
 			return __sync_lock_test_and_set(&ptr, p);
 		#endif
 	}
@@ -352,7 +352,7 @@
 			#else
 				return (T *)_InterlockedCompareExchange((volatile long *)&ptr, (long)(size_t)newValue, (long)(size_t)oldValue);
 			#endif
-		#elif defined(VD_COMPILER_CLANG)
+		#elif defined(VD_COMPILER_CLANG_OR_GCC)
 			return __sync_val_compare_and_swap(&ptr, oldValue, newValue);
 		#endif
 	}
