--- /home/ilm/Documents/GitHub/Altirra-4.50-test8-src/src/Kasumi/source/uberblit_resample.cpp	2025-09-14 23:26:06.572914100 +0200
+++ /home/ilm/Documents/GitHub/Altirra-4.50-test9-src/src/Kasumi/source/uberblit_resample.cpp	2026-04-15 05:41:51.771652300 +0200
@@ -212,7 +212,7 @@
 			{ kVDPixType_8,			false,	nsVDPixmap::kFilterSharpLinear,	CPUF_SUPPORTS_MMX,	RowFactorySharpLinear<VDResamplerSeparableTableRowStage8MMX> },
 			{ kVDPixType_8888,		false,	nsVDPixmap::kFilterSharpLinear,	CPUF_SUPPORTS_SSE2,	RowFactorySharpLinear<VDResamplerSeparableTableRowStageSSE2> },
 			{ kVDPixType_8888,		false,	nsVDPixmap::kFilterSharpLinear,	CPUF_SUPPORTS_MMX,	RowFactorySharpLinear<VDResamplerSeparableTableRowStageMMX> },
-#elif defined _M_AMD64
+#elif defined(_M_AMD64) && !defined(VD_COMPILER_GCC)
 			// AMD64
 			{ kVDPixType_8,			false,	nsVDPixmap::kFilterLinear,		CPUF_SUPPORTS_SSE2,	RowFactoryLinear<VDResamplerSeparableTableRowStage8SSE2> },
 			{ kVDPixType_8888,		false,	nsVDPixmap::kFilterLinear,		CPUF_SUPPORTS_SSE2,	RowFactoryLinear<VDResamplerSeparableTableRowStageSSE2> },
@@ -583,7 +583,7 @@
 		{ kVDPixType_8,			false,	nsVDPixmap::kFilterSharpLinear,	CPUF_SUPPORTS_MMX,	ColFactorySharpLinear<VDResamplerSeparableTableColStage8MMX> },
 		{ kVDPixType_8888,		false,	nsVDPixmap::kFilterSharpLinear,	CPUF_SUPPORTS_SSE2,	ColFactorySharpLinear<VDResamplerSeparableTableColStageSSE2> },
 		{ kVDPixType_8888,		false,	nsVDPixmap::kFilterSharpLinear,	CPUF_SUPPORTS_MMX,	ColFactorySharpLinear<VDResamplerSeparableTableColStageMMX> },
-#elif defined _M_AMD64
+#elif defined(_M_AMD64) && !defined(VD_COMPILER_GCC)
 		// AMD64
 		{ kVDPixType_8,			false,	nsVDPixmap::kFilterLinear,		CPUF_SUPPORTS_SSE2,	ColFactoryLinear<VDResamplerSeparableTableColStage8SSE2> },
 		{ kVDPixType_8888,		false,	nsVDPixmap::kFilterLinear,		CPUF_SUPPORTS_SSE2,	ColFactoryLinear<VDResamplerSeparableTableColStageSSE2> },
