--- /home/ilm/Documents/GitHub/Altirra-4.50-test8-src/src/ATIO/source/audioreaderflac.cpp	2024-01-17 06:17:46.568025700 +0100
+++ /home/ilm/Documents/GitHub/AltirraSDL/src/ATIO/source/audioreaderflac.cpp	2026-03-30 21:59:25.593069621 +0200
@@ -232,10 +232,12 @@
 
 #if VD_CPU_X86 || VD_CPU_X64
 	if constexpr (T_HaveLzcnt) {
-		#ifdef VD_COMPILER_CLANG
+		#if defined(VD_COMPILER_CLANG)
 			accumZeroes = [](uint32 v) __attribute__((target("lzcnt"))) {
 				return _lzcnt_u32(v);
 			}(bitAccum);
+		#elif defined(VD_COMPILER_GCC)
+			accumZeroes = __builtin_clz(bitAccum);
 		#else
 			accumZeroes = _lzcnt_u32(bitAccum);
 		#endif
@@ -243,7 +245,11 @@
 #endif
 	{
 		unsigned long oneBitPos;
+#ifdef _MSC_VER
 		_BitScanReverse(&oneBitPos, bitAccum);
+#else
+		oneBitPos = 31 - __builtin_clz(bitAccum);
+#endif
 
 #if VD_CPU_ARM64
 		// This is more optimal for ARM64, as the compiler doesn't realize
