--- /home/ilm/Documents/GitHub/Altirra-4.50-test8-src/src/ATTest/source/TestIO_Vorbis.cpp	2024-01-17 06:17:46.810326200 +0100
+++ /home/ilm/Documents/GitHub/Altirra-4.50-test9-src/src/ATTest/source/TestIO_Vorbis.cpp	2026-04-15 05:41:50.351073600 +0200
@@ -36,7 +36,7 @@
 		throw MyError("File too big");
 
 	vdblock<uint8> data((size_t)fileSize);
-	f.read(data.data(), (long)fileSize);
+	f.read(data.data(), (sint32)fileSize);
 	f.close();
 
 	vdautoptr dec { new ATVorbisDecoder };
@@ -101,7 +101,7 @@
 		throw MyError("File too big");
 
 	vdblock<uint8> data((size_t)fileSize);
-	f.read(data.data(), (long)fileSize);
+	f.read(data.data(), (sint32)fileSize);
 	f.close();
 
 	vdblock<sint16> buf(16384);
@@ -149,7 +149,7 @@
 		throw MyError("File too big");
 
 	vdblock<uint8> data((size_t)fileSize);
-	f.read(data.data(), (long)fileSize);
+	f.read(data.data(), (sint32)fileSize);
 	f.close();
 
 	vdblock<sint16> buf(16384);
