--- /home/ilm/Documents/GitHub/AltirraSDL/src/Altirra/source/trace.cpp	2026-03-31 14:47:31.345980122 +0200
+++ /home/ilm/Documents/GitHub/Altirra-4.50-test9-src/src/Altirra/source/trace.cpp	2026-04-15 05:41:48.331957100 +0200
@@ -16,7 +16,7 @@
 //	Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
 #include <stdafx.h>
-#include <cfloat>
+#include <float.h>
 #include <at/atcpu/history.h>
 #include "trace.h"
 
