--- /home/ilm/Documents/GitHub/Altirra-4.50-test8-src/src/h/vd2/system/zip.h	2025-09-14 23:26:06.287064200 +0200
+++ /home/ilm/Documents/GitHub/AltirraSDL/src/h/vd2/system/zip.h	2026-03-30 18:27:20.306983085 +0200
@@ -29,7 +29,11 @@
 // Rest in peace, Phil Katz.
 
 #include <string.h>
-#include <intrin.h>
+#ifdef _MSC_VER
+	#include <intrin.h>
+#elif defined(__x86_64__) || defined(__i386__)
+	#include <x86intrin.h>
+#endif
 #include <vd2/system/vdtypes.h>
 #include <vd2/system/binary.h>
 #include <vd2/system/error.h>
