--- /home/ilm/Documents/GitHub/Altirra-4.50-test8-src/src/h/vd2/system/binary.h	2025-06-13 03:48:25.188445700 +0200
+++ /home/ilm/Documents/GitHub/AltirraSDL/src/h/vd2/system/binary.h	2026-03-30 21:49:12.824248242 +0200
@@ -139,6 +139,13 @@
 						(((uint64)value & 0x000000000000FF00) << 40) +
 						(((uint64)value & 0x00000000000000FF) << 56));
 	}
+
+	inline uint32 VDRotateLeftU32(uint32 value, int bits) {
+		return (value << bits) | (value >> (32 - bits));
+	}
+	inline uint32 VDRotateRightU32(uint32 value, int bits) {
+		return (value >> bits) | (value << (32 - bits));
+	}
 #endif
 
 #if defined(__clang__) || defined(__GNUC__)
