--- /home/ilm/Documents/GitHub/AltirraSDL/src/ATCore/source/decmath.cpp	2026-04-06 09:18:44.937044464 +0200
+++ /home/ilm/Documents/GitHub/Altirra-4.50-test9-src/src/ATCore/source/decmath.cpp	2026-04-15 05:41:49.967016500 +0200
@@ -20,7 +20,7 @@
 //	archive for details.
 
 #include <stdafx.h>
-#include <cmath>
+#include <math.h>
 #include <at/atcore/decmath.h>
 
 bool ATDecFloat::SetDouble(double v) {
