--- /home/ilm/Documents/GitHub/Altirra-4.50-test8-src/src/Altirra/source/console.cpp	2026-02-16 06:23:56.613719400 +0100
+++ /home/ilm/Documents/GitHub/Altirra-4.50-test9-src/src/Altirra/source/console.cpp	2026-04-15 05:41:47.750572100 +0200
@@ -416,9 +416,9 @@
 
 void ATConsoleRequestFile(ATDebuggerRequestFileEvent& event) {
 	if (event.mbSave)
-		event.mPath = VDGetSaveFileName('dbgr', ATUIGetNewPopupOwner(), L"Select file to write for debugger command", L"All files\0*.*\0", nullptr);
+		event.mPath = VDGetSaveFileName("dbgr"_vdfcctypeid, ATUIGetNewPopupOwner(), L"Select file to write for debugger command", L"All files\0*.*\0", nullptr);
 	else
-		event.mPath = VDGetLoadFileName('dbgr', ATUIGetNewPopupOwner(), L"Select file to read for debugger command", L"All files\0*.*\0", nullptr);
+		event.mPath = VDGetLoadFileName("dbgr"_vdfcctypeid, ATUIGetNewPopupOwner(), L"Select file to read for debugger command", L"All files\0*.*\0", nullptr);
 }
 
 void ATInitUIPanes() {
